src/openklant/components/klantinteracties/models/validators.py

```diff
@@ -17,7 +19,147 @@
+                {
+                    "sub_identificator_van": _(
+                        "Voor de Identificator met CodeSoortObjectId = `vestigingsnummer` is het verplicht om"
+                        " een `sub_identifier_van` met CodeSoortObjectId = `kvk_nummer` te kiezen."
+                    )
+                }
+            )
+
+        if (
+            self.sub_identificator_van.partij_identificator_code_soort_object_id
+            != PartijIdentificatorCodeSoortObjectId.kvk_nummer.value
+        ):
+            raise ValidationError(
+                {
+                    "sub_identificator_van": _(
+                        "Het is alleen mogelijk om sub_identifier_vans te selecteren"
+                        " die CodeSoortObjectId = `kvk_nummer` hebben."
+                    )
+                }
+            )
+
+    def validate_unique_partij_identificator_locally(self):
+        """
+        Validation that a single Partij can only have a single partij_identificator_code_soort_object_id type locally
+        """
+        return  # TODO unlock after feedback
+        if (
+            self.partij
+            and self.queryset.filter(partij=self.partij)
+            .filter(
+                partij_identificator_code_soort_object_id=self.partij_identificator[
+                    "code_soort_object_id"
+                ]
+            )
+            .exists()
+        ):
+            raise ValidationError(
+                {
+                    "partij_identificator_code_soort_object_id": _(
+                        "Er is al een PartyIdentificator met dit CodeSoortObjectId = '%s' voor deze Partij."
+                        % (self.partij_identificator["code_soort_object_id"])
+                    )
+                }
+            )
+
+    def validate_unique_partij_identificator_globally(self):
+        """
+        Validation that a single partij_identifier combination occurs only once globally
+        """
+        filters = {
+            "partij_identificator_code_objecttype": self.partij_identificator[
+                "code_objecttype"
+            ],
+            "partij_identificator_code_soort_object_id": self.partij_identificator[
+                "code_soort_object_id"
+            ],
+            "partij_identificator_object_id": self.partij_identificator["object_id"],
+            "partij_identificator_code_register": self.partij_identificator[
+                "code_register"
+            ],
+        }
+
+        if self.sub_identificator_van is None:
+            filters["sub_identificator_van__isnull"] = True
+        else:
+            filters["sub_identificator_van"] = self.sub_identificator_van
+
+        if self.queryset.filter(**filters).exists():
+            raise ValidationError(
+                {
+                    "__all__": _(
+                        "`PartijIdentificator` moet uniek zijn, er bestaat er al een met deze gegevenscombinatie."
+                    )
+                }
+            )
```

Comment:
This means that the `invalid_params.name` in the API response will be `__all__`, while it should be `nonFieldErrors`

```suggestion
            raise ValidationError(
                _(
                        "`PartijIdentificator` moet uniek zijn, er bestaat er al een met deze gegevenscombinatie."
                 )
            )
```